controllers/service_controller_unit_test.go

```diff
@@ -0,0 +1,253 @@
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+const (
+
+	// Don't using the const here just because if you change the const, the behaviour
+	// will change, and things will not work as expected, keep it separate.
+
+	TestKuadrantDiscoveryAnnotationOASConfigMap = "discovery.kuadrant.io/oas-configmap"
+	TestKuadrantDiscoveryAnnotationOASPath      = "discovery.kuadrant.io/oas-path"
+	TestKuadrantDiscoveryAnnotationOASNamePort  = "discovery.kuadrant.io/oas-name-port"
+
+	PetStoreOAS = `
+openapi: "3.0.0"
+info:
+	title: "Petstore Service"
+	version: "1.0.0"
+servers:
+	- url: https://petstore.swagger.io/v1
+paths:
+	/pets:
+		get:
+			operationId: "getPets"
+			responses:
+				405:
+					description: "invalid input"`
+)
+
+func getReconciler() *reconcilers.BaseReconciler {
+
+	s := scheme.Scheme
+	err := appsv1.AddToScheme(s)
+	if err != nil {
+		return nil
+	}
+
+	CatOASConfigMap := &v1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "cat-oas",
+			Namespace: "test",
+		},
+		Data: map[string]string{
+			"openapi.yaml": PetStoreOAS,
+		},
+	}
+
+	DogOASConfigMap := &v1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "dog-oas",
+			Namespace: "test",
+		},
+		Data: map[string]string{
+			"openapi-invalid.yaml": PetStoreOAS,
+		},
+	}
+	// Objects to track in the fake client.
+	objs := []runtime.Object{CatOASConfigMap, DogOASConfigMap}
+
+	// Create a fake client to mock API calls.
+	cl := fake.NewFakeClient(objs...)
+	clientAPIReader := fake.NewFakeClient(objs...)
+	recorder := record.NewFakeRecorder(10000)
+
+	baseReconciler := reconcilers.NewBaseReconciler(cl, s, clientAPIReader, LOGTEST, recorder)
+	return baseReconciler
+}
+
+func getServiceReconciler() *ServiceReconciler {
+	return &ServiceReconciler{getReconciler()}
+}
+
+func getSampleService() *v1.Service {
+	return &v1.Service{
+		TypeMeta: metav1.TypeMeta{APIVersion: "v1", Kind: "Service"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        "test",
+			Namespace:   "test",
+			Labels:      map[string]string{KuadrantDiscoveryLabel: "true"},
+			Annotations: map[string]string{},
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{
+				{Port: 8080, Name: "test", Protocol: v1.ProtocolTCP, TargetPort: intstr.FromInt(3000)},
+				{Port: 10000, Name: "openapi", Protocol: v1.ProtocolTCP, TargetPort: intstr.FromInt(3000)},
+			},
+			Selector: map[string]string{"svc": "cats"},
+		},
+	}
+}
+
+// TestOasDiscoveryNoAnnotation test that check if no annotation nothing
+// happens at all
+func TestOasDiscoveryNoAnnotation(t *testing.T) {
+	serviceReconciler := getServiceReconciler()
+	svc := getSampleService()
+	log := serviceReconciler.Logger().WithValues("test", "foo")
```

Comment:
minor thing, but I think it is better to use a different logger on testing code. something like:
```go
import (
    ctrl "sigs.k8s.io/controller-runtime" 
)
log = ctrl.Log.WithName("test")

Reply:
Fixed.

Reply:
I do not see the fix

Reply:
I do now